### ISSUE TITLE
gzip uncompression error: "could not process image header unable to construct readers"

### ISSUE THREAD

**Issue by jtudelag** [kubevirt/containerized-data-importer#514]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
 /kind bug

Not sure if a bug or expected behaviour.

**What happened**:
Playing with CDI, trying to uncompress a gzip file in a volume, actually using the CDI release artifact,
which is a tar.gz. https://github.com/kubevirt/containerized-data-importer/archive/v1.3.0.tar.gz

> oc logs importer-cirros-wbqqd

I1102 15:47:14.047008       1 importer.go:32] Starting importer
I1102 15:47:14.047344       1 importer.go:37] begin import process
I1102 15:47:14.047369       1 dataStream.go:235] copying "https://github.com/kubevirt/containerized-data-importer/archive/v1.3.0.tar.gz" to "/data/disk.img"...                                                   
E1102 15:47:15.119924       1 importer.go:40] EOF
could not process image header
unable to construct readers
kubevirt.io/containerized-data-importer/pkg/importer.newDataStream
        /go/src/kubevirt.io/containerized-data-importer/pkg/importer/dataStream.go:128
kubevirt.io/containerized-data-importer/pkg/importer.NewDataStream
        /go/src/kubevirt.io/containerized-data-importer/pkg/importer/dataStream.go:103
kubevirt.io/containerized-data-importer/pkg/importer.CopyImage
        /go/src/kubevirt.io/containerized-data-importer/pkg/importer/dataStream.go:236
main.main
        /go/src/kubevirt.io/containerized-data-importer/cmd/cdi-importer/importer.go:38
runtime.main
        /gimme/.gimme/versions/go1.10.linux.amd64/src/runtime/proc.go:198
runtime.goexit
        /gimme/.gimme/versions/go1.10.linux.amd64/src/runtime/asm_amd64.s:2361
unable to create data stream
kubevirt.io/containerized-data-importer/pkg/importer.CopyImage
        /go/src/kubevirt.io/containerized-data-importer/pkg/importer/dataStream.go:238
main.main
        /go/src/kubevirt.io/containerized-data-importer/cmd/cdi-importer/importer.go:38
runtime.main
        /gimme/.gimme/versions/go1.10.linux.amd64/src/runtime/proc.go:198
runtime.goexit
        /gimme/.gimme/versions/go1.10.linux.amd64/src/runtime/asm_amd64.s:2361

**What you expected to happen**:

To have the file uncompressed inside the PV.

**How to reproduce it (as minimally and precisely as possible)**:

```
cat << EOF > pvc-cirros.yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: "cirros"
  labels:
    app: containerized-data-importer
  annotations:
    cdi.kubevirt.io/storage.import.endpoint: "https://github.com/kubevirt/containerized-data-importer/archive/v1.3.0.tar.gz"
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 10Gi
EOF
```

Create the PVC
```
oc create -f pvc-cirros.yaml
```

See the CDI Pod logs
```
oc logs  importer-cirros-XXX 
```

**Anything else we need to know?**:

**Environment**:
- CDI version (use `kubectl get deployments cdi-deployment -o yaml`):

```
- name: IMPORTER_IMAGE
          value: kubevirt/cdi-importer:v1.2.0
        - name: CLONER_IMAGE
          value: kubevirt/cdi-cloner:v1.2.0
        - name: UPLOADSERVER_IMAGE
          value: kubevirt/cdi-uploadserver:v1.2.0
        - name: PULL_POLICY
          value: IfNotPresent
        - name: UPLOADPROXY_SERVICE
          value: cdi-uploadproxy
```

- Kubernetes version (use `kubectl version`):
```
oc version
oc v3.9.40
kubernetes v1.9.1+a0ce1bc657
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://master1.4a26.internal:443
openshift v3.9.40
kubernetes v1.9.1+a0ce1bc657
```

- Cloud provider or hardware configuration:
- Install tools:
- Others:

Was palying with @scollier tutorial, used the cdi.sh script to install CDI. No kubevirt installed, just CDI.
https://github.com/scollier/kubevirt-tutorial

**Comment by jtudelag** [comment]
I have a use case where I need to populate volumes with data. This data might live in s3 buckets easily, so CDI seems to fit perfectly, but I would need CDI to uncompress the data in the volume. 

Now Im using this approach: https://medium.com/@jmarhee/using-initcontainers-to-pre-populate-volume-data-in-kubernetes-99f628cd4519

But I would like to do it with CDI, if thats possible and this is a valid use case for it;). 

Apparently now (from the logs) I can see it just download the file and copy it with the name disk.img:
´´´
I1102 15:42:56.839776       1 importer.go:32] Starting importer                                                                                                                                                    
I1102 15:42:56.840047       1 importer.go:37] begin import process                                                                                                                                                 
I1102 15:42:56.840064       1 dataStream.go:235] copying "https://github.com/openshift/origin/releases/download/v3.11.0/openshift-origin-client-tools-v3.11.0-0cbc58b-linux-64bit.tar.gz" to "/data/disk.img"...   
´´´

**Comment by awels** [comment]
@jtudelag keep an eye on #498 which will implement exactly what you are looking for. The current unpacking is only for when you have a VM disk image in a tar file.

**Comment by jtudelag** [comment]
thanks @awels, I will! This is awesome ;). Should I keep this open then? Or should I close it?

**Comment by awels** [comment]
@jtudelag Keep it open, I will associate this with the PR, once it gets merged it should close automatically.

**Comment by rollandf** [comment]
Solved by #498
